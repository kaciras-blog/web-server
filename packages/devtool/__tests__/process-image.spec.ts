import { expect, it, vi } from "vitest";
import { getAsset, runVite, testEntry } from "./test-utils";
import processImage from "../lib/plugin/process-image";

vi.mock("@kaciras-blog/media", () => ({
	createPresetCropper: () => vi.fn(async () => Buffer.alloc(3)),
}));

it("should crop the image", async () => {
	const bundle = await runVite({
		plugins: [
			processImage(),
			testEntry("import s from './test.png?size=IndexBannerM'; window.s=s"),
		],
	});

	expect(bundle.output[0].code).toMatchSnapshot();
	expect(getAsset(bundle, "test.png")).toHaveLength(3);
});
