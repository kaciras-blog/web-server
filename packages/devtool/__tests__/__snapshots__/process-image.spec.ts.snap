// Vitest Snapshot v1

exports[`should crop the image 1`] = `
"var s=\\"/test.png\\";window.s=s;
"
`;
